Реализованы контроллер, модели, сервис и репозиторий

mramazanov opened the pull request.

mramazanov committed:
Реализованы контроллер, модели, сервис и репозиторий

mramazanov committed:
Внесены правки в репозиторий, скрипт миграции и БД

gmnpr commented on src/main/resources/db/changelog/202503031105_init_tables.sql:
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE x6order.order (
+    id SERIAL PRIMARY KEY,
+    userId INT NOT NULL,
+    productId INT NOT NULL,
```
а если в заказе больше чем один продукт?)

gmnpr commented on src/main/java/com/javajabka/x6order/controller/OrderController.java:
```diff
@@ -0,0 +1,30 @@
+    private final OrderService orderService;
+
+    @PostMapping
+    @Operation(summary = "Создать пользователя")
```
пользователя?)

gmnpr commented on src/main/java/com/javajabka/x6order/exception/RestTemplateResponseErrorHandler.java:
```diff
@@ -0,0 +1,28 @@
+import java.io.IOException;
+import java.net.URI;
+
+public class RestTemplateResponseErrorHandler implements ResponseErrorHandler {
```
это бин. нужено добавть анотацию @Component

mramazanov committed:
1. Изменена структура таблицы order
2. Внесены изменения в репозиторий
3. Добавлено логирование logback
4. Внесены правки в тесты
5. Общие правки в code style

mramazanov committed:
Добавлен bean компонент для конфигурации RestTemplate

gmnpr commented on src/main/resources/db/changelog/202503031105_init_tables.sql:
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE x6order.order (
+    id SERIAL PRIMARY KEY,
+    userId INT NOT NULL,
+    products BIGINT[] NOT NULL,
```
нет :(
тебе нужна еще одна таблица и связь один ко многим

quantity тут к чему относится?) ко всему массиву, к кокнретному элементу массива?)

mramazanov replied:
Я тут изначально начал делать со второй таблицей со связью...Даже уверен был, но после некоторых твоих замечаний мне уже стало казаться что я накручиваю...

mramazanov committed:
1. Добавлена lombok генерация логера log4j2
2. Изменена схема БД
3. Изменен репозиторий

gmnpr commented on src/main/resources/db/changelog/202503031105_init_tables.sql:
```diff
@@ -0,0 +1,15 @@
+CREATE TABLE x6order.order (
+    id SERIAL PRIMARY KEY,
+    userId INT NOT NULL,
+    quantity INT NOT NULL,
```
эх, думаю тут косячок

gmnpr commented on src/main/java/com/javajabka/x6order/service/OrderService.java:
```diff
@@ -0,0 +1,59 @@
+        }
+    }
+
+    private void getProducts(OrderRequest orderRequest) {
```
это точно get? ты что-то получаешь?

gmnpr commented on src/main/java/com/javajabka/x6order/service/OrderService.java:
```diff
@@ -0,0 +1,59 @@
+        }
+    }
+
+    private void getUser(OrderRequest orderRequest) {
```
это точно get? ты что-то получаешь?

gmnpr commented on src/main/resources/application.properties:
```diff
@@ -1 +1,10 @@
+
+springdoc.swagger-ui.path=/swagger-ui
+
+logging.file.name=x6order_errors.log
```
зачем?

gmnpr commented on src/main/resources/logback-spring.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
а этот зачем?))

mramazanov committed:
1. Убрано логирование в файл
2. Поправлены имена методов rest запросов
3. Зменена схема БД

mramazanov committed:
Внесены правки в тесты

mramazanov committed:
Внесены правки в репозиторий и в модель OrderRequest

gmnpr commented on src/main/java/com/javajabka/x6order/repository/OrderRepository.java:
```diff
@@ -0,0 +1,71 @@
+        return params;
+    }
+
+    private MapSqlParameterSource[] productQuntityParams(final Long orderId, final OrderRequest orderRequest) {
```
Quantity
